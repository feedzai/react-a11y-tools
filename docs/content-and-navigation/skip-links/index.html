<!doctype html>
<html class="docs-version-current" lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.8">
<title data-react-helmet="true">Skip Links | react-a11y-tools</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" property="og:url" content="https://feedzai.github.io/react-a11y-tools/docs/content-and-navigation/skip-links"><meta data-react-helmet="true" name="docusaurus_locale" content="en"><meta data-react-helmet="true" name="docusaurus_version" content="current"><meta data-react-helmet="true" name="docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Skip Links | react-a11y-tools"><meta data-react-helmet="true" name="description" content="Provide ways for users to skip to important sections of our websites. It helps users using"><meta data-react-helmet="true" property="og:description" content="Provide ways for users to skip to important sections of our websites. It helps users using"><link data-react-helmet="true" rel="shortcut icon" href="/react-a11y-tools/img/favicon.svg"><link data-react-helmet="true" rel="canonical" href="https://feedzai.github.io/react-a11y-tools/docs/content-and-navigation/skip-links"><link data-react-helmet="true" rel="alternate" href="https://feedzai.github.io/react-a11y-tools/docs/content-and-navigation/skip-links" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://feedzai.github.io/react-a11y-tools/docs/content-and-navigation/skip-links" hreflang="x-default"><link rel="stylesheet" href="/react-a11y-tools/assets/css/styles.a411f444.css">
<link rel="preload" href="/react-a11y-tools/assets/js/runtime~main.93c78fde.js" as="script">
<link rel="preload" href="/react-a11y-tools/assets/js/main.6bb5e799.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#" class="skipToContent_1oUP">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/react-a11y-tools/"><div class="navbar__logo"><img src="/react-a11y-tools/img/favicon.svg" alt="React a11y tools" class="themedImage_1VuW themedImage--light_3UqQ"><img src="/react-a11y-tools/img/favicon.svg" alt="React a11y tools" class="themedImage_1VuW themedImage--dark_hz6m"></div><b class="navbar__title">React a11y tools</b></a></div><div class="navbar__items navbar__items--right"><a href="https://www.npmjs.com/package/@feedzai/react-a11y-tools" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link"><span>NPM<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a><a href="https://github.com/feedzai/react-a11y-tools" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link"><span>GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a><div class="toggle_71bT toggle_3Zt9 toggleDisabled_3cF-"><div class="toggleTrack_32Fl" role="button" tabindex="-1"><div class="toggleTrackCheck_3lV7"><span class="toggleIcon_O4iE">ðŸŒœ</span></div><div class="toggleTrackX_S2yS"><span class="toggleIcon_O4iE">ðŸŒž</span></div><div class="toggleTrackThumb_xI_Z"></div></div><input type="checkbox" class="toggleScreenReader_28Tw" aria-label="Switch between dark and light mode"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docs-wrapper docs-doc-page"><div class="docPage_31aa"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_35hR" type="button"></button><aside class="docSidebarContainer_3Kbt"><div class="sidebar_15mo"><nav class="menu thin-scrollbar menu_Bmed"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/react-a11y-tools/docs/getting-started">Getting Started</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">Feedback</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">Focus</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">Testing Utilities</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#">Content and Navigation</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/react-a11y-tools/docs/content-and-navigation/skip-links">Skip Links</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/react-a11y-tools/docs/content-and-navigation/semantic-headings">Semantic Headings</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">Hooks</a></li></ul></nav></div></aside><main class="docMainContainer_3ufF"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_3FnS"><div class="docItemContainer_33ec"><article><div class="theme-doc-markdown markdown"><header><h1>Skip Links</h1></header><h2 class="subtitle_sbI7">Provide ways for users to skip to important sections of our websites. It helps users using screen-readers to navigate easier and more efficiently.</h2><article class="browser_11iY" data-testid="fdz-js-docs-browser-window" tabindex="-1"><div class="browser__header_1GPe"><div class="buttons_22v8"><span class="dot_2OX7" style="background:#f25f58"></span><span class="dot_2OX7" style="background:#fbbe3c"></span><span class="dot_2OX7" style="background:#58cb42"></span></div><div class="browser__address-bar_2LAK"></div><div class="browser__menu-icon_1mIP"><div><span class="bar_2FYO"></span><span class="bar_2FYO"></span><span class="bar_2FYO"></span></div></div></div><div class="browser__body_2e2f"><button class="skipLinks__button_3f-n" data-testid="fdz-js-skip-links-target-button">Hey, start here!</button><div class="skipLinks__page_m1Rw"><div class="fdz-css-skip-links" data-testid="fdz-js-skip-links"><button role="link" type="button" class="fdz-css-skip-links__item">Skip to main content</button><button role="link" type="button" class="fdz-css-skip-links__item">Go to navigation menu</button></div><div class="skipLinks__container_3pg3"><header class="skipLinks__header_3LGS"><h1>Skip Links Demo</h1><nav class="skipLinks__target_1ez5" id="navigation-menu" tabindex="-1" data-testid="fdz-js-skip-links-nav"><ul class="skipLinks__list_1DZp"><li><a class="skipLinks__anchor_1-jk" href="#website">Link 1</a></li><li><a class="skipLinks__anchor_1-jk" href="#documentation">Link 2</a></li><li><a class="skipLinks__anchor_1-jk" href="#gitlab">Link 3</a></li></ul></nav></header><main class="skipLinks__target_1ez5" id="content" tabindex="-1" data-testid="fdz-js-skip-links-main"><h1 data-testid="fdz-js-heading" class="skipLinks__main-title_2Yxn">Bacon Ipsum</h1><p class="skipLinks__paragraph_3_6E"><span class="skipLinks__first-letter_XW7Z">B</span> eef ribs leberkas capicola cow swine turducken tri-tip meatball drumstick short ribs. Meatball boudin ribeye tri-tip pancetta, filet mignon kevin venison turducken pork loin. Porchetta buffalo kevin capicola tail pancetta shank corned beef bacon tongue beef ribs cow. Picanha ham hock ribeye sausage.<br><br>Chislic cupim pig, hamburger jerky ribeye andouille sausage buffalo. Tenderloin t-bone beef pancetta boudin short ribs pork loin kevin, spare ribs corned beef picanha filet mignon hamburger frankfurter ham hock. Prosciutto chuck chislic hamburger, pork loin meatloaf biltong tongue ham hock strip steak pork belly turkey venison boudin kevin. Sirloin capicola kevin ham spare ribs tri-tip swine leberkas pork loin cupim chislic jowl. Shoulder short loin pork loin, filet mignon biltong venison beef spare ribs tongue sausage. T-bone porchetta sausage picanha landjaeger shank venison beef bacon shankle rump.</p><a class="skipLinks__anchor_1-jk" id="website" href="https://github.com/feedzai/react-a11y-tools" target="_blank" rel="noopener noreferrer">Repository on Github</a><a class="skipLinks__anchor_1-jk" id="documentation" href="https://www.npmjs.com/package/@feedzai/react-a11y-tools" target="_blank" rel="noopener noreferrer">Package on npm</a></main></div></div></div></article><br><div class="admonition admonition-note alert alert--secondary"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="14" height="16" viewBox="0 0 14 16"><path fill-rule="evenodd" d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"></path></svg></span>How to</h5></div><div class="admonition-content"><ul><li>Press the <code>tab</code> key</li><li>A skip link will be shown on the top of the demo.</li><li>Press <code>enter</code></li><li>Focus will be set on the area of that <code>SkipLink</code>.</li></ul></div></div></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/feedzai/react-a11y-tools/docs/content-and-navigation/skip-links.mdx" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_2_ui" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_13-_"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/react-a11y-tools/docs/testing/keyboard-only"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Â« <!-- -->Keyboard Only</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/react-a11y-tools/docs/content-and-navigation/semantic-headings"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Semantic Headings<!-- --> Â»</div></a></div></nav></div></div></div></div></main></div></div></div>
<script src="/react-a11y-tools/assets/js/runtime~main.93c78fde.js"></script>
<script src="/react-a11y-tools/assets/js/main.6bb5e799.js"></script>
</body>
</html>