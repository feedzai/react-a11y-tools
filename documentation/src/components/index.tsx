export * from "./preview";
export * from "./Subtitle";
