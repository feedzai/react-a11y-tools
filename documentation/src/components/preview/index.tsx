import React, { ReactNode } from "react";

import styles from "./styles.module.scss";

interface Props {
	children: ReactNode;
	minHeight: number;
	url: string;
}

export function BrowserWindow({ children, minHeight }: Props) {
	return (
		<article
			className={styles.browser}
			style={{ minHeight }}
			data-testid="js-docs-browser-window"
			tabIndex={-1}
		>
			<div className={styles["browser__header"]}>
				<div className={styles.buttons}>
					<span className={styles.dot} style={{ background: "#f25f58" }} />
					<span className={styles.dot} style={{ background: "#fbbe3c" }} />
					<span className={styles.dot} style={{ background: "#58cb42" }} />
				</div>
				<div className={styles["browser__address-bar"]} />
				<div className={styles["browser__menu-icon"]}>
					<div>
						<span className={styles.bar} />
						<span className={styles.bar} />
						<span className={styles.bar} />
					</div>
				</div>
			</div>

			<div className={styles["browser__body"]}>{children}</div>
		</article>
	);
}
