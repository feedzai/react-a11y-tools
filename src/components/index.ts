/*
 * Please refer to the terms of the license
 * agreement.
 *
 * (c) 2021 Feedzai, Rights Reserved.
 */
export {
	MessagesAnnouncer,
	MessagesAnnouncerConsumer,
	useMessagesAnnouncer,
} from "./announcer/messages/index";
export { RouteAnnouncer } from "./announcer/route-announcer";
export { KeyboardOnly } from "./keyboard-only";
export * from "./focus-manager/index";
export { Heading, Level, Level as HeadingLevel, useHeadings } from "./semantic-headings";
export { SkipLinks } from "./skip-links";
export {
	RoverConsumer,
	RoverProvider,
	RoverContext,
	useFocus,
	useRover,
} from "./roving-tabindex/index";
export { VisuallyHidden, visuallyHiddenStyle } from "./visually-hidden";
