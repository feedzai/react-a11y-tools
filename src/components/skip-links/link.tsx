/**
 * This file is open-source. This means that it can be reproduced in whole
 * or in part, stored in a retrieval system transmitted in any form, or by
 * any means electronic with my prior permission as an author and owner
 * Please refer to the terms of the license agreement in the root of the project
 *
 * (c) 2021 joaodias.me
 */
import React, { FunctionComponent, useCallback, KeyboardEvent } from "react";
import "./styles.css";

export const SKIP_LINK_DEFAULT_PROPS: ISkipLink = {
	target: "#content",
	text: "Skip to main content",
	as: "link",
};

export interface ISkipLink {
	target: string;
	text: string;
	as?: "link" | "button";
}

/**
 * Skip Link to Main Content
 *
 * @param {ISkipLink} props
 * @returns {JSX.Element}
 */
export const SkipLink: FunctionComponent<ISkipLink> = ({ target, text, as }) => {
	const onKeyUp = useCallback(
		(event: KeyboardEvent<HTMLButtonElement>) => {
			if (event.key === "Enter" || event.key === " ") {
				const targetElement: HTMLElement | null = document.querySelector(target);

				targetElement?.focus();
			}
		},
		[target],
	);

	if (as === "button") {
		return (
			<button role="link" type="button" onKeyUp={onKeyUp} className="skip-links__item">
				{text}
			</button>
		);
	}
	return (
		<a href={target} className="skip-links__item">
			{text}
		</a>
	);
};

SkipLink.defaultProps = SKIP_LINK_DEFAULT_PROPS;

export default SkipLink;
