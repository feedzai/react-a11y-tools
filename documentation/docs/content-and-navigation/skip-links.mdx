---
sidebar_position: 1
---

import { DemoSkipLinks } from "./demos";
import { Subtitle, BrowserWindow, PropsTable } from "../../src/components/index";

# Skip Links

<Subtitle>
	Provide ways for users to skip to important sections of our websites. It helps users using
	screen-readers to navigate easier and more efficiently.
</Subtitle>

<BrowserWindow>
	<DemoSkipLinks />
</BrowserWindow>
<br />

:::note How to

- Press the `tab` key
- A skip link will be shown on the top of the demo.
- Press `enter`
- Focus will be set on the area of that `SkipLink`.
:::

<PropsTable name="SkipLinks" />
