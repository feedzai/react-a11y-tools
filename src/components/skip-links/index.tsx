/**
 * This file is open-source. This means that it can be reproduced in whole
 * or in part, stored in a retrieval system transmitted in any form, or by
 * any means electronic with my prior permission as an author and owner
 * Please refer to the terms of the license agreement in the root of the project
 *
 * (c) 2021 joaodias.me
 */
import React, { FunctionComponent } from "react";
import { ISkipLink, SkipLink, SKIP_LINK_DEFAULT_PROPS } from "./link";
export interface ISkipLinksProps {
	items?: ISkipLink[];
}

/**
 * Skip Links are links at the top of the page which jumps the user down to an anchor
 * or target at the beginning of the main content
 *
 * @param {ISkipLinksProps} props
 * @returns {JSX.Element}
 */
export const SkipLinks: FunctionComponent<ISkipLinksProps> = ({ items }) => {
	/**
	 * Renders a list of `SkipLink` components
	 *
	 * @returns {JSX.Element}
	 */
	function renderLinks() {
		const list =
			items && items.length > 0 ? (
				items.map((item) => <SkipLink key={item.target} {...item} />)
			) : (
				<SkipLink
					as={SKIP_LINK_DEFAULT_PROPS.as}
					target={SKIP_LINK_DEFAULT_PROPS.target}
					text={SKIP_LINK_DEFAULT_PROPS.text}
				/>
			);

		return (
			<div className="skip-links" data-testid="skip-links">
				{list}
			</div>
		);
	}

	return renderLinks();
};

SkipLinks.defaultProps = {
	items: [
		{
			...SKIP_LINK_DEFAULT_PROPS,
		},
	],
};
