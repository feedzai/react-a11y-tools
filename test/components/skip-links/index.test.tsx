/*
 * Please refer to the terms of the license
 * agreement.
 *
 * (c) 2021 Feedzai, Rights Reserved.
 */

/**
 * index.test.tsx
 *
 * @author Jo√£o Dias <joao.dias@feedzai.com>
 * @since 1.0.0
 */
import { screen, render, fireEvent, within, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import React from "react";
import { SkipLinks } from "../../../src/components/skip-links/index";
import { ISkipLink } from "../../../src/components/skip-links/link";
import { ISkipLinksProps } from "../../../src/components/skip-links";
import { SkipLink, SKIP_LINK_DEFAULT_PROPS } from "../../../src/components/skip-links/link";

const DemoContent = (props: ISkipLinksProps) => {
	return (
		<div className="wrapper">
			<SkipLinks {...props} />
			<main id="main-content" tabIndex={-1}>
				<p>Main Content Area</p>
			</main>
		</div>
	);
};

const defaultItems: ISkipLink[] = [
	{
		target: "#main-content",
		text: "Skip to Main Content",
		as: "button",
	},
	{
		target: "#sidebar",
		text: "Skip to Side Menu",
	},
];

describe("<SkipLinks>", () => {
	let props: ISkipLinksProps;

	beforeEach(() => {
		props = {
			items: defaultItems,
		};
	});

	it("should render a link by default", () => {
		const component = render(<SkipLinks />);
		expect(component).toMatchSnapshot();
	});

	it("should render a link even if items is undefined", () => {
		const component = render(<SkipLinks items={[]} />);
		expect(component).toMatchSnapshot();
	});

	it("should render a list of skip links", () => {
		const component = render(<SkipLinks {...props} />);
		expect(component).toMatchSnapshot();
	});

	describe("<SkipLink>", () => {
		it("should render without error", () => {
			const component = render(<SkipLink {...SKIP_LINK_DEFAULT_PROPS} />);
			expect(component).toMatchSnapshot();
		});

		describe("Placing focus on target", () => {
			let link: HTMLElement;

			beforeEach(() => {
				// Render the demo content
				const buttonName = props.items ? props.items[0].text : "";
				render(<DemoContent {...props} />);

				// Press the tab key on the body
				fireEvent.keyPress(document.body, { key: "Tab", code: 9, charCode: 9 });

				// Should show the first skip link
				const links = screen.getByTestId("skip-links");
				link = within(links).getByRole("link", { name: buttonName });
				expect(link).toBeVisible();
			});

			describe("Should place Focus", () => {
				afterEach(async () => {
					// The target should have focus
					await waitFor(() => {
						expect(screen.getByRole("main")).toHaveFocus();
					});
				});

				it("when using the ENTER key", () => {
					// Press enter on the first skip link
					fireEvent.keyUp(link, { key: "Enter", code: "Enter", charCode: 13 });
				});

				it("when using the SPACE key", () => {
					// Press space on the first skip link
					fireEvent.keyUp(link, { key: " ", code: "Space", charCode: 32 });
				});
			});

			it("should not place focus, when pressing any other key on a skip link", async () => {
				// Press a number of keys on the first skip link
				fireEvent.keyUp(link, { key: "a", code: "KeyA", charCode: 65 });
				fireEvent.keyUp(link, { key: "Control", code: "ControlLeft", charCode: 17 });
				fireEvent.keyUp(link, { key: "ArrowDown", code: "ArrowDown", charCode: 40 });

				// The target should not have focus, because the key isn't recognized as valid
				await waitFor(() => {
					expect(screen.getByRole("main")).not.toHaveFocus();
				});
			});
		});
	});
});
