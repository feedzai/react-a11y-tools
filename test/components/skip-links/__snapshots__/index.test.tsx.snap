// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkipLinks> <SkipLink> should render without error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="skip-links__item"
        href="#content"
      >
        Skip to main content
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="skip-links__item"
      href="#content"
    >
      Skip to main content
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<SkipLinks> should render a link by default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="skip-links"
        data-testid="skip-links"
      >
        <a
          class="skip-links__item"
          href="#content"
        >
          Skip to main content
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="skip-links"
      data-testid="skip-links"
    >
      <a
        class="skip-links__item"
        href="#content"
      >
        Skip to main content
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<SkipLinks> should render a link even if items is undefined 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="skip-links"
        data-testid="skip-links"
      >
        <a
          class="skip-links__item"
          href="#content"
        >
          Skip to main content
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="skip-links"
      data-testid="skip-links"
    >
      <a
        class="skip-links__item"
        href="#content"
      >
        Skip to main content
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<SkipLinks> should render a list of skip links 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="skip-links"
        data-testid="skip-links"
      >
        <button
          class="skip-links__item"
          role="link"
          type="button"
        >
          Skip to Main Content
        </button>
        <a
          class="skip-links__item"
          href="#sidebar"
        >
          Skip to Side Menu
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="skip-links"
      data-testid="skip-links"
    >
      <button
        class="skip-links__item"
        role="link"
        type="button"
      >
        Skip to Main Content
      </button>
      <a
        class="skip-links__item"
        href="#sidebar"
      >
        Skip to Side Menu
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
